ehrql/query_language.py

```diff
@@ -152,6 +154,16 @@ def if_null_then(self, other):
             default=self._cast(other),
         )
 
+    @staticmethod
```

Comment:
I don't think it makes sense to have these as static methods. We either want them as standalone functions in the `ehrql` namespace or as methods on the `ComparableFunctions` mixin.

I'm not quite sure what the best option is here though. Having them as methods would be nice from the point of view of not needing an extra import. But I can't quite think of a way of naming them where the resulting ehrQL makes sense. Maybe something like:
```
baseline_date = vaccination_date.or_lesser_of(infection_date, "2022-01-02")
```

But we can bikeshed that separately in Slack with a wider group I think.

Reply:
You're right, my thinking with making them static methods was that they are things that you can do to collections of PatientSeries or EventSeries but it's currently syntactically _meh_ to call them from an instance of one of those series. I now read that there's no prohibition on doing such a thing ( C#-brain strikes again!). I'll start up some bikeshedding in Slack once I've addressed some of the more obvious revisions